modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java

```diff
@@ -33,7 +33,7 @@
 
   @Override
   protected void configure() {
-    if (!ProfileRepository.getMode().equals(Mode.NONE)) {
+    if (!ProfileRepository.getRequestMode().equals(Mode.NONE)) {
```

Comment:
You will need to use jvm level setting here.

ProfileRepository.getGlobalProfileSetting().getMode()

Reply:
Done.